src/quo2/components/inputs/browser_input/view.cljs

```diff
@@ -0,0 +1,87 @@
+(ns quo2.components.inputs.browser-input.view
+  (:require [quo2.components.icon :as icon]
+            [quo2.components.inputs.browser-input.style :as style]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [react-native.platform :as platform]))
+
+(defn clear-icon
+  [blur? override-theme]
+  (if blur?
+    (colors/theme-colors colors/neutral-80-opa-30 colors/white-opa-10 override-theme)
+    (colors/theme-colors colors/neutral-40 colors/neutral-60 override-theme)))
+
+(defn- clear-button
+  [{:keys [on-press blur? override-theme]}]
+  [rn/touchable-opacity
+   {:accessibility-label :browser-input-clear-button
+    :style               style/clear-icon-container
+    :on-press            on-press}
+   [icon/icon :i/clear
+    {:color (clear-icon blur? override-theme)
+     :size  20}]])
+
+(defn cursor-color
+  [customization-color override-theme]
+  (colors/theme-colors (colors/custom-color customization-color 50)
+                       (colors/custom-color customization-color 60)
+                       override-theme))
+
+(defn placeholder-color
+  [state blur? override-theme]
+  (cond
+    (and blur? (= state :active))
+    (colors/theme-colors colors/neutral-80-opa-20 colors/white-opa-20 override-theme)
+
+    blur? ; state is default
+    (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-30 override-theme)
+
+    (= state :active)
+    (colors/theme-colors colors/neutral-30 colors/neutral-60 override-theme)
+
+    :else ; Not blur and state is default
+    (colors/theme-colors colors/neutral-40 colors/neutral-50 override-theme)))
+
+(def ^:private props-to-remove
+  [:cursor-color :placeholder-text-color :editable :on-change-text :on-focus
+   :on-blur :on-clear :value :disabled? :blur? :customization-color :override-theme])
+
+(defn browser-input
+  [{:keys [value]}]
+  (let [state       (reagent/atom :default)
+        set-active  #(reset! state :active)
+        set-default #(reset! state :default)
+        use-value?  (boolean value)]
+    (fn [{:keys [value disabled? blur? on-change-text customization-color
```

Comment:
value shouldn't be used , `default-value` should be used instead

Reply:
Alright

Reply:
@flexsurfer the `value` prop is not used by default but can be used when we want to implement clearing the input

Reply:
Oh, seen the implementation of `text-input` would change it to an uncontrolled input with ref to clear input when needed

Reply:
still we shouldn't manage value outside, it should be default-value, and value should be used only for clear-button button but it should be managed by component itself

Reply:
> still we shouldn't manage value outside, it should be default-value, and value should be used only for clear-button button but it should be managed by component itself

That is from the previous commit, I have removed it in the latest push I made to this branch.

See [here](https://github.com/status-im/status-mobile/pull/16039/commits/ca829e9329832c53786e6127d0d30c2e2cb9fdb1#diff-68f12f9bfa2d1b88a7ef49983ec2a8f870e56caa8b13cef8f62a52d4302be7b3L55)

Reply:
i mean there should be no value in params

Reply:
Oh I get you, would make that fix now

Reply:
@flexsurfer Updated the PR, moved value and the clear functionality into the component.